core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -58,27 +64,51 @@ public AtomicFileWriter(File f) throws IOException {
     }
 
     /**
-     * @param encoding
-     *      File encoding to write. If null, platform default encoding is chosen.
+     * @param encoding File encoding to write. If null, platform default encoding is chosen.
+     *
+     * @deprecated Use {@link #AtomicFileWriter(Path, Charset)}
      */
-    public AtomicFileWriter(File f, String encoding) throws IOException {
-        File dir = f.getParentFile();
-        try {
-            dir.mkdirs();
-            tmpFile = File.createTempFile("atomic",null, dir);
-        } catch (IOException e) {
-            throw new IOException("Failed to create a temporary file in "+ dir,e);
-        }
-        destFile = f;
-        if (encoding==null)
-            encoding = Charset.defaultCharset().name();
+    @Deprecated
+    public AtomicFileWriter(@Nonnull File f, @Nullable String encoding) throws IOException {
+        this(toPath(f), encoding == null ? Charset.defaultCharset() : Charset.forName(encoding));
+    }
+
+    /**
+     * Wraps potential {@link java.nio.file.InvalidPathException} thrown by {@link File#toPath()} in an
+     * {@link IOException} for backward compatibility.
+     *
+     * @param file
+     * @return the path for that file
+     * @see File#toPath()
+     */
+    private static Path toPath(@Nonnull File file) throws IOException {
```

Comment:
NIT static functions normally come after constructors in the style guide.

Reply:
+1 I would also put this method to a utility class. we have lots of other cases where it is required